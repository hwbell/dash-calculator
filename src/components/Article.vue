<template>
  <div class="article-component shadow-card">
    <div class="space-all-row">
      <div class="left-all-col">
        <p class="article-title">{{ article.title }}</p>
        <p>{{ article.publishedAt.slice(0, 10) }}</p>
      </div>
      <img
        class="article-image"
        v-bind:src="article.urlToImage"
        alt="article image"
      />
    </div>
    <p>
      <span class="article-author" v-if="article.author"
        >{{ article.author }}, </span
      ><span v-if="article.source.name"> from {{ article.source.name }}</span>
    </p>
    <p v-if="article.content" class="article-description">
      <span>{{
        article.content.slice(
          0,
          -(article.content.length - article.content.indexOf("["))
        )
      }}</span>
    </p>
    <a v-bind:href="article.url" target="_blank" alt="article link"
      >&nbsp;&nbsp;&nbsp;read full article</a
    >
  </div>
</template>

<script>
export default {
  name: "Article",
  props: {
    article: Object
  },
};
</script>

<!-- Add "scoped" attribute to limit CSS to this component only -->
<style scoped>
.article-component {
  background-color: white;
  width: 100%;
  margin: 20px 0;
  padding: 20px;
  border-radius: 2px;
  /* border: 1px solid rgb(207, 206, 206); */
}
.article-image {
  /* width: 100px; */
  height: 100px;
  border-radius: 10px;
}
.article-title {
  color: black;
  font-weight: bolder;
  font-size: 14pt;
}
.article-author {
  color: rgb(63, 63, 63);
  font-weight: bolder;
  font-size: 12pt;
}
.article-description {
  color: black;
  font-size: 12pt;
}
</style>
