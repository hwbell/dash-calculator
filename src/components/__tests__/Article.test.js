import { mount } from '@vue/test-utils'
import Article from '../Article.vue'

const article = {
  source: { id: null, name: 'Global News' },
  author: 'Curt Anderson And Brian Mahoney',
  title: 'U.S. Memorial Day crowds small, subdued amid coronavirus pandemic - Global News',
  description: '<ol><li>U.S. Memorial Day crowds small, subdued amid coronavirus pandemic  Global News\r\n' +
    '</li><li>US marks quiet Memorial Day as coronavirus deaths near 100,000  Al Jazeera English\r\n' +
    '</li><li>What Is the Difference Between Memorial Day and Veterans Day?  Newswe…',
  url: 'http://globalnews.ca/news/6981600/coronavirus-memorial-day-warnings/',
  urlToImage: 'https://shawglobalnews.files.wordpress.com/2020/05/1000-32.jpeg?quality=70&strip=all&w=720&h=379&crop=1',
  publishedAt: '2020-05-25T11:01:48Z',
  content: 'Americans settled for small processions and online tributes instead of parades Monday as they observed Memorial Day in the shadow of the pandemic, which forced communities to honour the nation’s mili… [+5855 chars]'
}

describe('Article', () => {

  let wrapper;
  beforeEach(() => {
    wrapper = mount(Article, {
      propsData: {
        article
      }
    })
  })

  test('is a Vue instance', () => {
    expect(wrapper.isVueInstance()).toBeTruthy()
  })

  test('renders correctly', () => {
    expect(wrapper.element).toMatchSnapshot()
  })

  test('renders a button for each object in the array map', () => {
    expect(wrapper.findAll('.article-component').length).toBe(1)
  })

  test('receives and uses the props', () => {
    expect(wrapper.find('.article-title').text()).toBe(article.title);
    expect(wrapper.find('.article-image').attributes().src).toBe(article.urlToImage)
  })
})