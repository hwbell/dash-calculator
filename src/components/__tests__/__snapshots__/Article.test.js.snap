// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Article renders correctly 1`] = `
<div
  class="article-component shadow-card"
>
  <div
    class="space-all-row"
  >
    <div
      class="left-all-col"
    >
      <p
        class="article-title"
      >
        U.S. Memorial Day crowds small, subdued amid coronavirus pandemic - Global News
      </p>
       
      <p>
        2020-05-25
      </p>
    </div>
     
    <img
      alt="article image"
      class="article-image"
      src="https://shawglobalnews.files.wordpress.com/2020/05/1000-32.jpeg?quality=70&strip=all&w=720&h=379&crop=1"
    />
  </div>
   
  <p>
    <span
      class="article-author"
    >
      Curt Anderson And Brian Mahoney, 
    </span>
    <span>
       from Global News
    </span>
  </p>
   
  <p
    class="article-description"
  >
    <span>
      Americans settled for small processions and online tributes instead of parades Monday as they observed Memorial Day in the shadow of the pandemic, which forced communities to honour the nation’s mili… 
    </span>
  </p>
   
  <a
    alt="article link"
    href="http://globalnews.ca/news/6981600/coronavirus-memorial-day-warnings/"
    target="_blank"
  >
       read full article
  </a>
</div>
`;
